src/content/1.6/code/agda/snippet35.agda

```diff
@@ -1,3 +1,9 @@
-sumToProd : Either (a × b) (a × c) → (a × Either b c)
-sumToProd (Left (x , y))  = x , Left y
-sumToProd (Right (x , z)) = x , Right z
+sumToProd : Either (A × B) (A × C) → A × Either B C
+sumToProd e = case e of λ where
```

Comment:
I did this to make the Agda code closer to the original Haskell code, but then I added the (perhaps more standard) implementation you provided an alternative.